[#173] Null class logger not logging to logger output

Eclipse-Dominator opened the pull request:
Fixes #173 

There is a check when a null class logger is created. 
However, AB3 currently do not actually use null class loggers.

Let's replace the null class check with assertion instead as per YAGNI principe.

damithc reviewed (changes requested):
The commit message raises some questions in the reader's mind: 
* What is a 'null class logger'?
* What is the 'incorrect method'? Why it is incorrect?

So, although the commit itself is small, looks like it needs a more detailed commit message.

damithc commented on src/main/java/seedu/address/commons/core/LogsCenter.java:
```diff
@@ -53,9 +53,7 @@ public static Logger getLogger(String name) {
-        if (clazz == null) {
-            return Logger.getLogger("");
-        }
+        assert clazz != null;
```
I remember we use a different method (rather that `assert`) to check for null parameters elsewhere in AB3. If that is the case, we can do the same, for consistency.

damithc reviewed (changes requested):
The current commit message is more informative than before. But a few more details are still missing.

Something like the below? Tweak as you see fit.
```
Add a null check to LogCenter.getLogger(Class<T>)

LogCenter.getLogger(Class<T> clazz) creates a Logger named
after the given clazz. If the clazz is null, it is supposed to
create a Logger with an empty String as the name. 

The current implementation calls Logger.getLogger("")
instead of LogCenter.getLogger(""), which results in the Logger
object not having the handlers that are added inside the
LogCenter.getLogger(String) method. This means log messages
written using such a logger will not appear in the log files.

However, the current codebase has no use case for passing a
null clazz to this method. As per the YAGNI principle,
we should not implement that ability until we actually need it.

Hence, instead of correcting the said error, let's add an assertion
to ensure null is never passed to this method, eliminating the need
for the buggy code path in the first place. 
```

Eclipse-Dominator committed:
Add a null check to LogCenter.getLogger(Class<T>)

LogCenter.getLogger(Class<T> clazz) creates a Logger named
after the given class. If the clazz is null, it is supposed to
create a Logger with an empty String as the name.

The current implementation calls Logger.getLogger("")
instead of LogCenter.getLogger(""), which results in the Logger
object not having the handlers that are added inside the
LogCenter.getLogger(String) method. This means log messages
written using such a logger will not appear in the log files.

However, the current codebase has no use case for passing a
null object to this method. As per the YAGNI principle,
we should not implement that ability until we actually need it.

Hence, instead of correcting the said error, let's add an assertion
to ensure null is never passed to this method, eliminating the need
for the buggy code path in the first place.

Eclipse-Dominator commented:
I have replaced certain instance of clazz with class in the commit since clazz is a term to replace class in java to avoid using preserved keyword.
